src/averageAges.js

```diff
@@ -37,7 +46,13 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => (person.sex === 'f')
+  && (withChildren ? people.find(children => children.mother === person.name)
+    : true));
+
+  const ageOfDeath = women.map(woman => woman.died - woman.born);
```

Comment:
you're getting an array from here, so it should better be called in the plural. I would come up with sth like `ages`

Reply:
Done! :D